default.nix

```diff
@@ -0,0 +1,251 @@
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages)
+  );
+
+  renderBooklet = booklet:
+    let
+      addPage = content: pageData: {
+        content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+          %****************************************************************
+          %*
+          %* ${pageData.sourceHeader}
+          %*
+          %****************************************************************
+        '' + ''
+          %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+          ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}
+          %%BeginPageSetup
+          /pgsave save def
+          %%EndPageSetup
+          ${if pageData ? isLandscape then "landscapePage" else "portraitPage"} begin
+          ${lib.optionalString (pageData ? drawFooter) "${toString content.nextFooterIdx} drawFooter"}
+          ${pageData.content}
+          end
+          pgsave restore
+          showpage
+        '';
+        nextFooterIdx = content.nextFooterIdx + (if pageData ? drawFooter then 1 else 0);
+        nextPgIdx = content.nextPgIdx + 1;
+      };
+      initialContent = {
+        content = ''
+          %!PS-Adobe-3.0
+          %%Orientation: Portrait
+          %%Pages: ${toString (builtins.length booklet.pages)}
+          %%EndComments
+          %%BeginSetup
+          ${toString (dependencyContent (booklet.pages))}%%EndSetup
+
+          %************************************************************************
+          %************************************************************************
+          %*
+          %* Section Three: Page Rendering
+          %*
+          %************************************************************************
+          %************************************************************************
+
+        '';
+        nextPgIdx = 1;
+        nextFooterIdx = 1;
+      };
+      finalContent = builtins.foldl' addPage initialContent booklet.pages;
+    in
+    pkgs.writeTextFile {
+      name = booklet.name;
+      text = finalContent.content + ''
+        %%EOF
+      '';
+    };
+  checkSinglePage = page: renderBooklet {
+    name = "test-single-page.ps";
+    pages = [ page ];
+  };
+in
+stdenv.mkDerivation {
+  name = "codex32${shortId}";
+
+  nativeBuildInputs = if doPdfGeneration then [ ghostscript ] else [ ];
+
+  phases = [ "buildPhase" ] ++ lib.optionals doCheck [ "checkPhase" ];
+
+  buildPhase = ''
+    set -e
+
+    FULL_BW="${renderBooklet fullBooklet}"
+
+    # Copy output Postscript into place
+    mkdir "$out"
+    cd "$out"
+    cp "$FULL_BW" SSS32.ps
+    # Patch to include version
+    sed -i 's/(revision \(.*\))/(revision \1${shortId})/' ./SSS32.ps
+    # Produce PDF, if requested.
+    ${lib.optionalString doPdfGeneration "ps2pdf -dPDFSETTINGS=/prepress SSS32.ps"}
+  '';
+
+  checkPhase = toString
+    (map
+      (page: "ghostscriptTest ${checkSinglePage page}")
```

Comment:
There is another issue with this design.  Because `checkPhase` contains references to the single pages, those single pages must be built, whether or not they will be checked.  In this case I suggest taking the unusual step of conditionally excluding the entire contents of the `checkPhase` value if `doCheck` is false.

Alternatively you could replace `fullBooklet.pages` with `[]` when `doCheck` is false.